### PR TITLE
Rework how palette is defined

### PR DESCRIPTION
**Pull request by isabela-pf**
As mentioned in #252, the Spyder team has been working on [updating the default dark and light themes in Spyder](https://github.com/spyder-ide/ux-improvements/issues/13), some of which is based on QDarkStyle. We wanted to give back our relevant work of restructuring how colors develop the palette, and this PR starts that process.

Here's what is changed in this PR:
- palette.py becomes darkpalette.py (anticipating the need for a lightpalette.py in the future). darkpalette.py is now linked to colorsystem.py to define the pre-existing constants.
- Adds colorsystem.py, a file that lists a full range of grays and blues (similar to current QDarkStyle) as a foundation for the initial palette and to support more flexible palette changes in the future.
- Adds colorreference.md in the docs to provide some context for these changes. The color samples folder holds the images used in these docs.

From the user side, the palette should still look about the same.

This type of work is not something I have a lot of experience with, so please forgive any errors. I'm absolutely open to changes.

I can't thank @ccordoba12 enough for helping me structure and correct this PR!

### CODE REVIEW

**Review by dpizetta — CHANGES_REQUESTED** [review]
Hi @isabela-pf, we have converted all the docs of this project to RST format, so we can add more things like tables what is missing in MD format. So, I think would be a good idea to keep this doc as RST too, if you don't mind. You can run the text on a website/tool to convert it.  Tks

File: qdarkstyle/palette.py [hunk 1]
```diff
@@ -21,15 +21,15 @@ class DarkPalette(object):
 
     COLOR_SELECTION_LIGHT = '#148CD2'
     COLOR_SELECTION_NORMAL = '#1464A0'
-    COLOR_SELECTION_DARK = '#14506E'
+    COLOR_SELECTION_DARK = '#14506E'"""
```
Comment (dpizetta) [inline qdarkstyle/palette.py]: @isabela-pf  you can go ahead and remove those colors as comments, so we have a clean code. :)

**Review by dpizetta — CHANGES_REQUESTED** [review]

**Comment by isabela-pf** [comment]
Thanks for the feedback! I think I did everything you asked in the last commit, but let me know if there's anything else. 

(Also thanks to @steff456 for troubleshooting my git adventures this time!)

File: qdarkstyle/darkpalette.py [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
-    COLOR_FOREGROUND_NORMAL = '#AAAAAA'
-    COLOR_FOREGROUND_DARK = '#787878'
+    COLOR_FOREGROUND_LIGHT = Gray.B120
+    COLOR_FOREGROUND_NORMAL = Gray.B80
+    COLOR_FOREGROUND_DARK = Gray.B60
 
-    COLOR_SELECTION_LIGHT = '#148CD2'
-    COLOR_SELECTION_NORMAL = '#1464A0'
-    COLOR_SELECTION_DARK = '#14506E'
+    COLOR_SELECTION_LIGHT = Blue.B60`
```
Comment (ccordoba12) [inline qdarkstyle/darkpalette.py:22]: You left a back tick at the end of this line. Please remove it.

**Review by ccordoba12 — COMMENTED** [review]
Great work @isabela-pf! I think this is ready, except for my comment below.

> I can't thank @ccordoba12 enough for helping me structure and correct this PR!

Thanks for your kind words! I think you got it almost right yourself, except for some minor details.

**Comment by isabela-pf** [comment]
Back tick removed Thanks for catching that!

**Review by ccordoba12 — APPROVED** [review]
Thanks @isabela-pf, this looks good to me now.

**Comment by ccordoba12** [comment]
@dpizetta, what do you think?

**Comment by ccordoba12** [comment]
Pinging @dpizetta again. I hope to hear from you soon, else I'll go ahead and merge.

**Comment by dpizetta** [comment]
Sorry guys, busy days.

I will make some changes to fix some problems when I run the process to generate all the resources, I do the merge in my machine until today night. As the process of generating those resources is not automatized yet, I would recommend we use the develop branch to PR's, so we can keep the fully functional code in the master (for Python and C++). 

Thanks for both

**Comment by dpizetta** [comment]
The old and new palette for dark theme after the compilation.

image

**Review by dpizetta — APPROVED** [review]

**Comment by dpizetta** [comment]
Merge done 👍

**Comment by isabela-pf** [comment]
Thank you so much! ❤️

**Comment by ccordoba12** [comment]
Thanks @dpizetta!

**Pull request closed without merging**